mips/pci.c

```diff
@@ -192,32 +170,27 @@ static void pci_bus_dump(pci_bus_t *pcibus) {
               'A' + pcidev->pin - 1, pcidev->irq);
 
     for (int i = 0; i < pcidev->nbars; i++) {
-      pci_bar_t *bar = &pcidev->bar[i];
-      pm_addr_t addr = bar->addr;
-      size_t size = bar->size;
+      resource_t *bar = &pcidev->bar[i];
       char *type;
 
-      if (addr & PCI_BAR_IO) {
-        addr &= ~PCI_BAR_IO_MASK;
+      if (bar->r_type == RT_IOPORTS) {
         type = "I/O ports";
       } else {
-        if (addr & PCI_BAR_PREFETCHABLE)
-          type = "Memory (prefetchable)";
-        else
-          type = "Memory (non-prefetchable)";
-        addr &= ~PCI_BAR_MEMORY_MASK;
+        type = (bar->r_flags & RF_PREFETCHABLE) ? "Memory (prefetchable)"
+                                                : "Memory (non-prefetchable)";
       }
       kprintf("%s Region %d: %s at %p [size=$%x]\n", devstr, i, type,
-              (void *)addr, (unsigned)size);
+              (void *)bar->r_start, (unsigned)(bar->r_end - bar->r_start + 1));
     }
   }
 }
 
-static pci_bus_t pci_bus[1];
+extern pci_bus_device_t gt_pci;
 
 void pci_init() {
   kmalloc_init(mp, 1, 1);
-  pci_bus_enumerate(pci_bus);
-  pci_bus_assign_space(pci_bus, MALTA_PCI0_MEMORY_BASE, PCI_IO_SPACE_BASE);
-  pci_bus_dump(pci_bus);
+
+  pci_bus_enumerate(&gt_pci);
```

Comment:
`pci_bus_enumerate` finds all devices behind this PCI bus, and creates corresponding `pci_device`s. So we'll probably want to call `enumerate` for all buses that are present in the system. Either we'll need a list of all PCI buses for that, or we'll be enumerating new buses when a PCI-PCI bridge driver is bound to a bridge (and thus a new bus is discovered).